tests/Go/PhpUnit/ClassIsNotWovenConstraint.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/*
+ * Go! AOP framework
+ *
+ * @copyright Copyright 2011, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Go\PhpUnit;
+
+use \PHPUnit_Framework_Constraint as Constraint;
+use \ReflectionClass;
+use Go\Instrument\PathResolver;
+
+/**
+ * Asserts that class is not woven.
+ */
+class ClassIsNotWovenConstraint extends Constraint
+{
+    /**
+     * @var array
+     */
+    private $configuration;
+
+    public function __construct(array $configuration)
+    {
+        parent::__construct();
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function matches($other)
+    {
+        $filename = (new ReflectionClass($other))->getFileName();
+        $suffix   = substr($filename, strlen(PathResolver::realpath($this->configuration['appDir'])));
+
+        return
+            !file_exists($this->configuration['cacheDir'] . $suffix)
+            &&
+            !file_exists($this->configuration['cacheDir'] . '/_proxies' . $suffix);
```

Comment:
If both file exists -> it is woven. 
If none of file exists -> it is not woven.

If one exists and one does not -> there has been an error and assert must fail. Therefore, || should not be used. 

In regards to expressions -> you have to have in mind that I am 16+ years in development and these kind of expressions are "human readable" for me.

As I said already, it is your library and I will try to follow your style and correct code, even when I disagree. But you have to cut me some slack sometimes - I do have my own style, habits and so on... these things will happen.